common/app/model/readerRevenue/BannerDeploy.scala

```diff
@@ -29,10 +29,20 @@ object ReaderRevenueRegion {
     }
   }
 
-  def getBucketKey(region: ReaderRevenueRegion): String = {
-    contributionsBannerDeployLogKey + "-" + region.name + ".json"
+  def getBucketKey(region: ReaderRevenueRegion, banner: BannerType): String = {
+      banner match {
+        case ContributionsBanner => createBucketKeyForRegion(contributionsBannerDeployLogKey, region)
+        case SubscriptionsBanner => createBucketKeyForRegion(subscriptionsBannerDeployLogKey, region)
+      }
+  }
+
+  def createBucketKeyForRegion(bucketKey: String, region: ReaderRevenueRegion): String = {
+    bucketKey + "-" + region.name + ".json"
   }
 
   val allRegions: List[ReaderRevenueRegion] = List(UK, US, AU, ROW)
 }
 
+sealed trait BannerType { val name: String }
+case object ContributionsBanner extends BannerType { val name = "contributions-banner" }
```

Comment:
I wonder these case objects should have a few more properties and you wouldn't need to do the pattern matches everywhere.  For example, you put the deploy log path as a property and you'd no longer need to do

```
banner match {
        case ContributionsBanner => createBucketKeyForRegion(contributionsBannerDeployLogKey, region)
        case SubscriptionsBanner => createBucketKeyForRegion(subscriptionsBannerDeployLogKey, region)
      }
```

It would become:

`createBucketKeyForRegion(banner.logKey, region)`

(or similar)

Reply:
If we did do something like this then maybe they could live in separate file